src/diagrams/gantt/ganttDb.spec.js

```diff
@@ -172,4 +172,18 @@ describe('when using the ganttDb', function () {
     expect(tasks[2].startTime).toEqual(moment('2013-01-15', 'YYYY-MM-DD').toDate())
     expect(tasks[2].endTime).toEqual(moment('2013-01-17', 'YYYY-MM-DD').toDate())
   })
+  it('should ignore weekends', function () {
+    ganttDb.setDateFormat('YYYY-MM-DD')
+    ganttDb.setExcludes('weekends 2019-02-06')
+    ganttDb.addSection('testa1')
+    ganttDb.addTask('test1', 'id1,2019-02-01,1d')
```

Comment:
If this would be:

```javascript
ganttDb.addTask('test1', 'id1,2019-02-01,8d')
```

You'd expect

```
 expect(tasks[1].startTime).toEqual(moment('2019-02-13', 'YYYY-MM-DD').toDate())
```

And your code would return 2019-01-11

Some more fringe test cases are needed here.

Reply:
Added more tests.